views/login.go

```diff
@@ -20,104 +21,60 @@ type LoginTemplate struct {
 	MsgType string
 }
 
-// LogoutFunc Implements the logout functionality.
-// Will delete the session information from the cookie store
-func (v *Views) LogoutFunc(w http.ResponseWriter, r *http.Request) {
-	session, err := v.cookie.Get(r, v.conf.SessionCookieName)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
-
-	session.Values["user"] = user.User{}
-	session.Options.MaxAge = -1
-	err = session.Save(r, w)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
-	// TODO Don't call env in this function have an initialiser
-	// then fetch from that store?
-	endpoint := v.conf.LogoutEndpoint
-	if endpoint == "" {
-		endpoint = "/"
-	}
-	http.Redirect(w, r, endpoint, http.StatusFound)
-}
-
 // LoginFunc implements the login functionality, will
 // add a cookie to the cookie store for managing authentication
-func (v *Views) LoginFunc(w http.ResponseWriter, r *http.Request) {
+func (v *Views) LoginFunc(c echo.Context) error {
 	var err error
 
-	session, _ := v.cookie.Get(r, v.conf.SessionCookieName)
+	session, _ := v.cookie.Get(c.Request(), v.conf.SessionCookieName)
 	// We're ignoring the error here since sometimes the cookies keys change, and then we
 	// can overwrite it instead
 
-	switch r.Method {
+	switch c.Request().Method {
 	case "GET":
 		// Data for our HTML template
 		context := v.getData(session)
 
 		// Check if there is a callback request
-		callbackURL, err := url.Parse(r.URL.Query().Get("callback"))
+		callbackURL, err := url.Parse(c.QueryParam("callback"))
 		if err == nil && strings.HasSuffix(callbackURL.Host, v.conf.BaseDomainName) && callbackURL.String() != "" {
 			context.Callback = callbackURL.String()
 		}
 		// Check if authenticated
 		if context.User.Authenticated {
-			http.Redirect(w, r, context.Callback, http.StatusFound)
-			return
+			//http.Redirect(w, r, context.Callback, http.StatusFound)
+			return c.Redirect(http.StatusFound, context.Callback)
 		}
 
-		data := LoginTemplate{
-			Version: v.conf.Version,
-			Message: "",
-			MsgType: "",
-		}
-
-		err = v.template.RenderNoNavsTemplate(w, data, templates.LoginTemplate)
-		//err = v.tpl.ExecuteTemplate(w, "login", context)
-		if err != nil {
-			log.Printf("login failed to exec tmpl: %+v", err)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-		}
-		return
+		return v.template.RenderTemplate(c.Response(), context, templates.LoginTemplate, templates.NoNavType)
 	case "POST":
 		// Parsing form to struct
-		err = r.ParseForm()
+		err = c.Request().ParseForm()
```

Comment:
This isn't necessary - per https://pkg.go.dev/net/http#Request.FormValue, `FormValue` will call `ParseForm` if needed.